### PR TITLE
Implemented spine based ordering while importing EPUB file to pressbook.

### PR DESCRIPTION
**Pull request by hakkim-pits**
We have fixed this issue.We have changed the file" inc/modules/import/epub/class-epub201.php "

https://github.com/pressbooks/pressbooks/issues/442

### CODE REVIEW

File: inc/modules/import/epub/class-epub201.php [hunk 1]
```diff
@@ -554,4 +580,41 @@ protected function kneadHref( \DOMDocument $doc, $type, $href ) {
...
+				continue;
+			}
+
+			$this->manifest[ $id ] = [
+										'type' => $type,
+										'herf' => $href,
+									];
+		}
+
+		return $this->manifest;
```
Comment (connerbw) [inline inc/modules/import/epub/class-epub201.php]: Return value is redundant.

File: inc/modules/import/epub/class-epub201.php [hunk 2]
```diff
@@ -554,4 +580,41 @@ protected function kneadHref( \DOMDocument $doc, $type, $href ) {
...
 		return $doc;
 	}
+
+
+		/**
+		 * Parse manifest with type 'application/xhtml+xml' to array
+		 *
+		 * @param \SimpleXMLElement $xml
+		 * @return array()
+		 */
```
Comment (connerbw) [inline inc/modules/import/epub/class-epub201.php]: Spacing is a bit off.

File: inc/modules/import/epub/class-epub201.php [hunk 3]
```diff
@@ -554,4 +580,41 @@ protected function kneadHref( \DOMDocument $doc, $type, $href ) {
...
+				}
+			}
+
+						// Skip
+			if ( 'application/xhtml+xml' !== $type ) {
+				continue;
+			}
+
+			$this->manifest[ $id ] = [
+										'type' => $type,
```
Comment (connerbw) [inline inc/modules/import/epub/class-epub201.php]: Spacing is a bit off.

File: inc/modules/import/epub/class-epub201.php [hunk 4]
```diff
@@ -554,4 +580,41 @@ protected function kneadHref( \DOMDocument $doc, $type, $href ) {
 
 		return $doc;
 	}
+
+
+		/**
+		 * Parse manifest with type 'application/xhtml+xml' to array
+		 *
+		 * @param \SimpleXMLElement $xml
+		 * @return array()
```
Comment (connerbw) [inline inc/modules/import/epub/class-epub201.php]: If you are modifying `$this->manifest` then there is no need to return it.

File: inc/modules/import/epub/class-epub201.php [hunk 5]
```diff
@@ -47,6 +47,14 @@ class Epub201 extends Import {
 
 
 	/**
+	 * Array of manifest with tytpe application/xhtml+xml
```
Comment (greatislander) [inline inc/modules/import/epub/class-epub201.php]: Minor typo here, should be `type`

Comment (hakkim-pits) [reply inc/modules/import/epub/class-epub201.php]: Done.

File: inc/modules/import/epub/class-epub201.php (hunk 2 above)
Comment (hakkim-pits) [reply inc/modules/import/epub/class-epub201.php]: Done.Changed it.

File: inc/modules/import/epub/class-epub201.php (hunk 3 above)
Comment (hakkim-pits) [reply inc/modules/import/epub/class-epub201.php]: Done.Changed it.

File: inc/modules/import/epub/class-epub201.php (hunk 4 above)
Comment (hakkim-pits) [reply inc/modules/import/epub/class-epub201.php]: Removed the return value. We had done it considering method chaining in future implementations if needed.

File: inc/modules/import/epub/class-epub201.php (hunk 1 above)
Comment (hakkim-pits) [reply inc/modules/import/epub/class-epub201.php]: Removed the return value. We had done it considering method chaining in future implementations if needed.

Comment (connerbw) [reply inc/modules/import/epub/class-epub201.php]: Cool idea. You would have to return `$this` for fluent methods however. Unless I'm misunderstanding? Anyways, looks good.

**Pull request merged**